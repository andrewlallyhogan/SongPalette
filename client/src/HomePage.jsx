import React, { useState, useEffect } from 'react';
import { useHistory } from 'react-router-dom';


function HomePage() {

  const history = useHistory();
  
  function loginPage() {
    history.push('/loginPage');
  }

  function createAcctPage() {
   history.push('/createAccount');
  }

  return (
    <div className='homeContainer' data-testid='homeContainer'>
      <div className='leftInterface'>
        <img
          className='leftImage'
          src='https://i.imgur.com/2fgUkIk.png'
          alt='shops'
        />
      </div>

      <div className='middleInterface'>
        <img
          className='middleImage'
          src='https://i.imgur.com/AHFcYv1.png'
          alt='shop-logo'
        />
          <img
            className='neighborImage'
            src='https://i.imgur.com/pD5v9ZC.png'
            alt='Neighbor'
          />
        <h2>"The go to source for small businesses in your neighborhood."</h2>
        <h2 className='quoteSource'>-Mr. Rogers</h2>
        <div>
          <img
            onClick={() => loginPage()}
            className='loginImage'
            src='https://i.imgur.com/rvjQK1R.png'
            alt='login'
          />
        </div>
        <div>
          <img
            onClick={() => createAcctPage()}
            className='createAccountImage'
            src='https://i.imgur.com/hpMvW3E.png'
            alt='create account'
          />
        </div>
      </div>
      <div className='rightInterface'>
        <img
          className='rightImage'
          src='https://i.imgur.com/4rNsynC.jpg'
          alt='shops'
        />
      </div>
    </div>
  );
}

export default HomePage;
