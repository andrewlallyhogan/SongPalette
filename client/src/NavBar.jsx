import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';

function NavBar() {
  const history = useHistory();

  return (
    <div className='navBar' data-testid='navbar'>
      {/* left element (directs to home page) */}
      <div className='leftNav'>
        <img
          onClick={() => history.push('/')}
          className='leftImage'
          src='https://i.imgur.com/AHFcYv1.png'
          alt='Logo'
        />
      </div>

      {/* spacer to create even distance between elements in nav bar */}
      <div className='spacer'></div>

      {/* middle element */}
      <div className='middleNeighbor'>
        <img
          className='siteFontLogo'
          src='https://i.imgur.com/pD5v9ZC.png'
          alt='Neighbor'
        />
      </div>

      {/* spacer to create even distance between elements in nav bar */}
      <div className='spacer'></div>

      {/* right elements direct to about us, team and user information. User information needs to be made to only be clickable after login */}
      <div className='rightNav'>
        <img
          onClick={() => history.push('/team')}
          className='teamImage'
          src='https://i.imgur.com/FFrUPpK.png'
          alt='Team'
        />
        |{' '}
        <img
          onClick={() => history.push('/aboutus')}
          className='aboutNeighborImage'
          src='https://i.imgur.com/su5pf5g.png'
          alt='About Neighbors'
        />
        |{' '}
        <img
          onClick={() => history.push('/userinformation')}
          className='userInfo'
          src='https://i.imgur.com/VWTc4IU.png'
          alt='User Information'
        />
      </div>
    </div>
  );
}

export default NavBar;
