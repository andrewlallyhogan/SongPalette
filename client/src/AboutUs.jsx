import React from 'react';
import NavBar from './NavBar';


// About Us page 
// Import Navigation bar component 

function AboutUs() {
  return (
    <div className='aboutUsPage'>
      <div>
        <NavBar />
      </div>

      <div className='aboutUsContainer'>
        <div className='aboutUsTextContainer'>
          <img id='aboutUsLogo' src='https://i.imgur.com/UWrIPFU.png'></img>
          <br />
          <br />
          <p>
            Welcome to Neighbor! We are an online platform that bridges the gap
            between shoppers and their local businesses. Our mission is to make
            it easier for people to discover and support their local businesses,
            while also providing a convenient shopping experience. We have an
            extensive directory of local merchants, offering a variety of goods
            and services. Whether you are looking for a unique gift, a special
            treat, or a one-of-a-kind item, you will be sure to find it on
            Neighbor. With special discounts, promotions, and offers, Neighbor
            is the perfect solution for all of your shopping needs. Shop local
            and support your community today! <br /> <br />
            Shopping at small businesses is important because it helps to
            strengthen local economies and create jobs. Small businesses are
            often the backbone of communities, providing goods and services that
            may not be available from larger retailers. Shopping at small
            businesses also helps to keep money in the local economy, since
            small businesses are more likely to reinvest their profits in the
            local economy. Additionally, small businesses often offer unique and
            personalized products and services that cannot be found elsewhere.
            <br /> <br />
            Sign up with Neighbor and start supporting your community today.
          </p>
        </div>
      </div>
    </div>
  );
}

export default AboutUs;
