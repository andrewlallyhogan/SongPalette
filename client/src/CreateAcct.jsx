import React, { useState, useEffect } from 'react';
import NavBar from './NavBar';
import { useHistory } from 'react-router-dom';
import axios from 'axios';

function CreateAcct({ setDefaultEmail, defaultEmail }) {
  const history = useHistory();

  const handleSetDefaultEmail = () => {
    event.preventDefault();

    console.log('Create User Button Clicked');
    // setDefaultEmail(event.target[2].value)
  };
  // Spacing below was done using manual spacers. Should be done via divs in SCSS.

  return (
    <div>
      <NavBar />
      <div>
        <h2 className='createAccount'>creating an account is easy</h2>
      </div>

      {/* hold both user and merchant login */}
      <div className='loginContainer'>
        <div className='consMerchContainer'>
          {/* user login */}
          <section className='formSection'>
            {/* header for user login */}
            <div className='userAlignment'>
              <strong>User</strong>
            </div>
            <br></br>

            {/* TEMPLATE HERE */}
            {/* <label htmlFor='firstName' >last name: </label>
              <input type='text' name='firstName' id='firstName'/> */}
            {/* TEMPLATE END HERE */}

            {/* form for user */}
            {/* ADD STYLING CLASSNAMES BACK IN LATER */}
<div className='innerAlign'>
            <form method='POST' action='http://localhost:3000/createUser'>
              <label htmlFor='firstName'>First Name: </label>
              <input type='text' name='firstName' id='firstName' />
              <br />

              <label htmlFor='lastName'>Last Name: </label>
              <input type='text' name='lastName' id='lastName' />
              <br />

              <label htmlFor='email'>Email: </label>
              <input type='text' name='email' id='email' />
              <br />

              <label htmlFor='password'>Password: </label>
              {/* CHANGE TYPE TO PASSWORD ONCE DONE W TESTING AND DEVELOPMEN */}
              <input type='password' name='password' id='password' />
              <br />

              <label htmlFor='zipCode'>Zip Code: </label>
              <input type='text' name='zipCode' id='zipCode' />
              <br />

              <input id='formButton' type='submit' value='submit' />
            </form>
            </div>
          </section>

          {/* merchant accoutn creation */}
          {/* merchantEmail,
          merchantPassword,
          merchantName,
          typeOfMerchant,
          merchantAddress,
          merchantZipCode,
          description,
          image, */}

          <div className='merchantLogin'>
            {/* header for user login */}
            <div className='merchantAlignment'>
              <strong>Merchant</strong>
            </div>
            <br></br>
<div className='innerAlign'>
            <form method='post' action='http://localhost:3000/createMerchant'>
              <label htmlFor='merchantEmail'>Email: </label>
              <input type='text' name='merchantEmail' id='merchantEmail' />
              <br />

              <label htmlFor='merchantPassword'>Password: </label>
              {/* CHANGE TYPE TO PASSWORD AFTER TESING AND DEVELOPMEN */}
              <input
                type='password'
                name='merchantPassword'
                id='merchantPassword'
              />
              <br />

              <label htmlFor='merchantName'>Business Name: </label>
              <input type='text' name='merchantName' id='merchantName' />
              <br />

              <label htmlFor='typeOfMerchant'>Business Type: </label>
              <input type='text' name='typeOfMerchant' id='typeOfMerchant' />
              <br />

              <label htmlFor='merchantAddress'> Address: </label>
              <input type='text' name='merchantAddress' id='merchantAddress' />
              <br />

              <label htmlFor='merchantZipCode'>Zip Code: </label>
              <input type='text' name='merchantZipCode' id='merchantZipCode' />
              <br />

              <label htmlFor='description'>Description: </label>
              <input type='text' name='description' id='description' />
              <br />

              <label htmlFor='image'>Image: </label>
              <input type='text' name='image' id='image' />

              <br />
              <input id='formButton' type='submit' value='submit' />
              <br />
              <br />
            </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}
export default CreateAcct;
