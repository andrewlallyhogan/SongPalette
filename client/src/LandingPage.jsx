import React, { useState } from 'react';
import NavBar from './NavBar';
import MerchantCards from './merchantCards';

// Imported mock data from MerchantCards.jsx
// Created a form and button to search for merchants by zip codes. 
function LandingPage(props) {
  console.log('LANDING PAGE props: ', props)
  const [searchZipcode, setSearchZipcode] = useState(''); 

  return (
    
      <div className='landingPage'>
        <NavBar />

        <div className = 'searchBar'>

          <form>
            <label>
              Zip Code:
              <input
              type='text'
              name='searchZipcode'
              value={searchZipcode}
              onChange={(event) => {setSearchZipcode(event.target.value)}}/>
            </label>
          </form>

        </div>
      <section className='merchantWrapper'>
      <MerchantCards merchantData={props.merchantData} searchZipcode={searchZipcode} />
    </section>
    </div>
  );
}

export default LandingPage;
