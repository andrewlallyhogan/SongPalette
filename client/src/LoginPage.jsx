import React, { useState, useEffect } from 'react';
import NavBar from './NavBar';
import { useHistory } from 'react-router-dom';
import axios from 'axios';
function LoginPage(props) {
  const history = useHistory();
  
  const handleLogin = () => {
    console.log('hi')
    axios({
      method: 'get',
      url: 'http://localhost:3000/allMerchants',
    })
      .then((responseData) => {
        console.log(' All Merchants: ', responseData.data)
        setMerchantData(responseData.data)
    })
    props.setNeedMerchantData(props.needMerchantData)
  }

  return (
    <div>
      <NavBar />
      <div>
        <h2 className='login'>Login</h2>
      </div>
      <div className='loginContainer'>
        {/* hold both user and merchant login */}
        
        <div className='consMerchContainer'>
          {/* user login */}
          <div className='userLogin'>
            <div className='userAlignment'>
              <strong>User</strong>
            </div>
            <br />
            {/* USER LOGIN FORM GOES HERE */}
            
            <div className='innerAlign'>
            <form
              onSubmit={handleLogin}
              className='userForm'
              method='post'
              action='http://localhost:3000/userLogin'
            >
              {/* <form className='userForm' onSubmit={handleUserLogin}*/}
              <div className='userLogin'>
                {/* USER EMAIL INPUT */}
                <label htmlFor='email'>Email: </label>
                <input type='text' name='email' id='email' />

                {/* <label htmlFor='userEmailLogin'>
                  Email:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
                  <input
                    type='text'
                    id='userEmailLogin'ss
                    value={userEmailLogin}
                    onChange={(event) =>
                      setUserEmailLogin(event.target.value)
                    }
                  />
                </label> */}
              </div>

              {/* USER PASSWORD INPUT */}
              <div className='userPW'>
                <label htmlFor='password'>Password: </label>
                <input type='password' name='password' id='password' />

                {/* <label htmlFor='userPasswordLogin'>
                  Password:
                  <input
                    type='text'
                    id='userPasswordLogin'
                    value={userPasswordLogin}
                    onChange={(event) => {
                      setUserPasswordLogin(event.target.value);
                    }}
                  />
                </label> */}
              </div>
              <input
                name='userSubmit'
                id='formButton'
                type='submit'
                value='submit'
              />
            </form>
            </div>
          </div>

          {/* MERCHANT LOGIN GOES HERE */}
          <div className='merchantLogin'>
            <div className='merchantAlignment'>
              <strong>Merchant</strong>
            </div>
            <br />
            <div className='innerAlign'>
            <form
              onSubmit={handleLogin}
              className='merchantForm'
              method='post'
              action='http://localhost:3000/merchantLogin'
            >
              {/* MERCHANT EMAIL LOGIN */}
              <div className='merchantUsername'>
                <label htmlFor='memail'>Email: </label>
                <input type='text' name='memail' id='memail' />

                {/* <label htmlFor='merchantUsernameLogin'>
                  Email: &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
                <input 
                type='text' 
                id='merchantEmailLogin'
                value={merchantEmailLogin}
                onChange={(event) => {
                  setMerchantEmailLogin(event.target.value)
                }}
                />
                </label> */}
              </div>

              {/* MERCHANT PASSWORD LOGIN */}
              <div className='merchantPassword'>
                <label htmlFor='mpassword'>Password: </label>
                <input type='password' name='mpassword' id='mpassword' />

                {/* <label htmlFor='merchantPasswordLogin'>
                  Password:
                  <input
                    type='text'
                    id='merchantPasswordLogin'
                    value={merchantPasswordLogin}
                    onChange={(event) => {
                      setMerchantPasswordLogin(event.target.value);
                    }}
                  />
                </label> */}
              </div>

              <input id='merchantFormButton' type='submit' value='submit' />
            </form>
            </div>
          </div>

        </div>
      </div>
    </div>
  );
}

export default LoginPage;
