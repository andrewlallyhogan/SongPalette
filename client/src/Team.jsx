import React from 'react';
import NavBar from './NavBar';

function Team() {
  return (
    <div className='teamPage' data-testid='team'>
      <div>
        <NavBar />
      </div>

      <div className='allTeamPhotos'>
        <div className='teamBox'>
        <div className='teamHeader'>CURRENT DEVELOPER TEAM</div>
        <div className='zach'>
          <img
            className='teamPhoto'
            src='https://i.imgur.com/W3fbp8y.png'
          ></img>
          <div className='teamText'>
            <p>
              Camille Salter  <br />
              <br />
              Backend Engineer & Tester
            </p>
            <a href='https://github.com/CamSalter'>Github</a>
          </div>
        </div>
        <div className='zach'>
          <img
            className='teamPhoto'
            src='https://i.imgur.com/TrzpIlP.png'
          ></img>
          <div className='teamText'>
            <p>
              Andrew Hogan <br />
              <br />
              Fullstack & Testing Engineer
            </p>
            <a href='https://github.com/andrewlallyhogan'>Github</a>
          </div>
        </div>
        <div className='chase'>
          <img
            className='teamPhoto'
            src='https://i.imgur.com/jCmd23c.png'
          ></img>
          <div className='teamText'>
            <p>
              Eshaan Joshi <br />
              <br />
              Frontend Engineer
            </p>
            <a href='https://github.com/eshaan32'>Github</a>
          </div>
        </div>

        <div className='danny'>
          <img
            className='teamPhoto'
            src='https://i.imgur.com/rb8rMhH.png'
          ></img>
          <div className='teamText'>
            <p>
              Greg Jenner <br />
              <br />
              Backend & Database Engineer
            </p>
            <a href='https://github.com/gregjenner'>Github</a>
          </div>
        </div>

        <div className='sakura'>
          <img
            className='teamPhoto'
            src='https://i.imgur.com/1XXAny6.png'
          ></img>
          <div className='teamText'>
            <p>
              Jason Goldman <br />
              <br />
              Fullstack Engineer
            </p>
            <a href='https://github.com/Trablink'>Github</a>
          </div>
        </div>
        </div>
      </div>
      
      <div className='allTeamPhotos'>
        <div className='teamBox'>
        <div className='teamHeader'>ORIGINAL DEVELOPERS</div>
        <div className='zach'>
          <img
            className='teamPhoto'
            src='https://i.imgur.com/A0Q4L5l.png'
          ></img>
          <div className='teamText'>
            <p>
              Zach Hall <br />
              <br />
              Network Ninja
            </p>
            <a href='https://github.com/zachh85'>Github</a>
          </div>
        </div>

        <div className='chase'>
          <img
            className='teamPhoto'
            src='https://i.imgur.com/kXsSP3w.png'
          ></img>
          <div className='teamText'>
            <p>
              Chase Benjamin <br />
              <br />
              Digital Data Detective
            </p>
            <a href='https://github.com/chasebenj'>Github</a>
          </div>
        </div>

        <div className='danny'>
          <img
            className='teamPhoto'
            src='https://i.imgur.com/5JzRNX2.png'
          ></img>
          <div className='teamText'>
            <p>
              Danny Zheng <br />
              <br />
              Automation Wizard
            </p>
            <a href='https://github.com/dannyzheng159'>Github</a>
          </div>
        </div>

        <div className='sakura'>
          <img
            className='teamPhoto'
            src='https://i.imgur.com/DHJlkOx.png'
          ></img>
          <div className='teamText'>
            <p>
              Sakura Akiyama-Bowden <br />
              <br />
              Server Wrangler
            </p>
            <a href='https://github.com/sakurakiyama'>Github</a>
          </div>
          </div>
        </div>
      </div>
    </div>
    
  );
}

export default Team;
