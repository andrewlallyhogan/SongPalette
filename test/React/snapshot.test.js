import renderer from 'react-test-renderer';
import React from 'react';
import NavBar from '../../client/src/NavBar';
import CreateAcct from '../../client/src/CreateAcct';
import HomePage from '../../client/src/HomePage';
import Team from '../../client/src/Team';
import LoginPage from '../../client/src/LoginPage';
import LandingPage from '../../client/src/LandingPage';
import AboutUs from '../../client/src/AboutUs';

// stay away from merchant data/card user information

describe('NavBar component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<NavBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('CreateAcct component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<CreateAcct />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Homepage component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<HomePage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Team component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Team />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('LoginPage component', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<LoginPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('LandingPage component', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<LandingPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('AboutUs component', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<AboutUs />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
