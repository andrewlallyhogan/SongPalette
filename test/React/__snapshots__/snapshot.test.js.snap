// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUs component should match snapshot 1`] = `
<div
  className="aboutUsPage"
>
  <div>
    <div
      className="navBar"
      data-testid="navbar"
    >
      <div
        className="leftNav"
      >
        <img
          alt="Logo"
          className="leftImage"
          onClick={[Function]}
          src="https://i.imgur.com/AHFcYv1.png"
        />
      </div>
      <div
        className="spacer"
      />
      <div
        className="middleNeighbor"
      >
        <img
          alt="Neighbor"
          className="siteFontLogo"
          src="https://i.imgur.com/pD5v9ZC.png"
        />
      </div>
      <div
        className="spacer"
      />
      <div
        className="rightNav"
      >
        <img
          alt="Team"
          className="teamImage"
          onClick={[Function]}
          src="https://i.imgur.com/FFrUPpK.png"
        />
        |
         
        <img
          alt="About Neighbors"
          className="aboutNeighborImage"
          onClick={[Function]}
          src="https://i.imgur.com/su5pf5g.png"
        />
        |
         
        <img
          alt="User Information"
          className="userInfo"
          onClick={[Function]}
          src="https://i.imgur.com/VWTc4IU.png"
        />
      </div>
    </div>
  </div>
  <div
    className="aboutUsContainer"
  >
    <div
      className="aboutUsTextContainer"
    >
      <img
        id="aboutUsLogo"
        src="https://i.imgur.com/UWrIPFU.png"
      />
      <br />
      <br />
      <p>
        Welcome to Neighbor! We are an online platform that bridges the gap between shoppers and their local businesses. Our mission is to make it easier for people to discover and support their local businesses, while also providing a convenient shopping experience. We have an extensive directory of local merchants, offering a variety of goods and services. Whether you are looking for a unique gift, a special treat, or a one-of-a-kind item, you will be sure to find it on Neighbor. With special discounts, promotions, and offers, Neighbor is the perfect solution for all of your shopping needs. Shop local and support your community today! 
        <br />
         
        <br />
        Shopping at small businesses is important because it helps to strengthen local economies and create jobs. Small businesses are often the backbone of communities, providing goods and services that may not be available from larger retailers. Shopping at small businesses also helps to keep money in the local economy, since small businesses are more likely to reinvest their profits in the local economy. Additionally, small businesses often offer unique and personalized products and services that cannot be found elsewhere.
        <br />
         
        <br />
        Sign up with Neighbor and start supporting your community today.
      </p>
    </div>
  </div>
</div>
`;

exports[`CreateAcct component matches the snapshot 1`] = `
<div>
  <div
    className="navBar"
    data-testid="navbar"
  >
    <div
      className="leftNav"
    >
      <img
        alt="Logo"
        className="leftImage"
        onClick={[Function]}
        src="https://i.imgur.com/AHFcYv1.png"
      />
    </div>
    <div
      className="spacer"
    />
    <div
      className="middleNeighbor"
    >
      <img
        alt="Neighbor"
        className="siteFontLogo"
        src="https://i.imgur.com/pD5v9ZC.png"
      />
    </div>
    <div
      className="spacer"
    />
    <div
      className="rightNav"
    >
      <img
        alt="Team"
        className="teamImage"
        onClick={[Function]}
        src="https://i.imgur.com/FFrUPpK.png"
      />
      |
       
      <img
        alt="About Neighbors"
        className="aboutNeighborImage"
        onClick={[Function]}
        src="https://i.imgur.com/su5pf5g.png"
      />
      |
       
      <img
        alt="User Information"
        className="userInfo"
        onClick={[Function]}
        src="https://i.imgur.com/VWTc4IU.png"
      />
    </div>
  </div>
  <div>
    <h2
      className="createAccount"
    >
      creating an account is easy
    </h2>
  </div>
  <div
    className="loginContainer"
  >
    <div
      className="consMerchContainer"
    >
      <section
        className="formSection"
      >
        <div
          className="userAlignment"
        >
          <strong>
            User
          </strong>
        </div>
        <br />
        <form
          action="http://localhost:3000/createUser"
          method="POST"
        >
          <label
            htmlFor="firstName"
          >
            First Name: 
          </label>
          <input
            id="firstName"
            name="firstName"
            type="text"
          />
          <br />
          <label
            htmlFor="lastName"
          >
            Last Name: 
          </label>
          <input
            id="lastName"
            name="lastName"
            type="text"
          />
          <br />
          <label
            htmlFor="email"
          >
            Email: 
          </label>
          <input
            id="email"
            name="email"
            type="text"
          />
          <br />
          <label
            htmlFor="password"
          >
            Password: 
          </label>
          <input
            id="password"
            name="password"
            type="password"
          />
          <br />
          <label
            htmlFor="zipCode"
          >
            Zip Code: 
          </label>
          <input
            id="zipCode"
            name="zipCode"
            type="text"
          />
          <br />
          <input
            id="formButton"
            type="submit"
            value="submit"
          />
        </form>
      </section>
      <div
        className="merchantLogin"
      >
        <div
          className="merchantAlignment"
        >
          <strong>
            Merchant
          </strong>
        </div>
        <br />
        <form
          action="http://localhost:3000/createMerchant"
          method="post"
        >
          <label
            htmlFor="merchantEmail"
          >
            Email: 
          </label>
          <input
            id="merchantEmail"
            name="merchantEmail"
            type="text"
          />
          <br />
          <label
            htmlFor="merchantPassword"
          >
            Password: 
          </label>
          <input
            id="merchantPassword"
            name="merchantPassword"
            type="password"
          />
          <br />
          <label
            htmlFor="merchantName"
          >
            Business Name: 
          </label>
          <input
            id="merchantName"
            name="merchantName"
            type="text"
          />
          <br />
          <label
            htmlFor="typeOfMerchant"
          >
            Business Type: 
          </label>
          <input
            id="typeOfMerchant"
            name="typeOfMerchant"
            type="text"
          />
          <br />
          <label
            htmlFor="merchantAddress"
          >
             Address: 
          </label>
          <input
            id="merchantAddress"
            name="merchantAddress"
            type="text"
          />
          <br />
          <label
            htmlFor="merchantZipCode"
          >
            Zip Code: 
          </label>
          <input
            id="merchantZipCode"
            name="merchantZipCode"
            type="text"
          />
          <br />
          <label
            htmlFor="description"
          >
            Description: 
          </label>
          <input
            id="description"
            name="description"
            type="text"
          />
          <br />
          <label
            htmlFor="image"
          >
            Image: 
          </label>
          <input
            id="image"
            name="image"
            type="text"
          />
          <br />
          <input
            id="formButton"
            type="submit"
            value="submit"
          />
          <br />
          <br />
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Homepage component matches the snapshot 1`] = `
<div
  className="homeContainer"
  data-testid="homeContainer"
>
  <div
    className="leftInterface"
  >
    <img
      alt="shops"
      className="leftImage"
      src="https://i.imgur.com/2fgUkIk.png"
    />
  </div>
  <div
    className="middleInterface"
  >
    <img
      alt="shop-logo"
      className="middleImage"
      src="https://i.imgur.com/AHFcYv1.png"
    />
    <img
      alt="Neighbor"
      className="neighborImage"
      src="https://i.imgur.com/pD5v9ZC.png"
    />
    <h2>
      "The go to source for small businesses in your neighborhood."
    </h2>
    <h2
      className="quoteSource"
    >
      -Mr. Rogers
    </h2>
    <div>
      <img
        alt="login"
        className="loginImage"
        onClick={[Function]}
        src="https://i.imgur.com/rvjQK1R.png"
      />
    </div>
    <div>
      <img
        alt="create account"
        className="createAccountImage"
        onClick={[Function]}
        src="https://i.imgur.com/hpMvW3E.png"
      />
    </div>
  </div>
  <div
    className="rightInterface"
  >
    <img
      alt="shops"
      className="rightImage"
      src="https://i.imgur.com/4rNsynC.jpg"
    />
  </div>
</div>
`;

exports[`LoginPage component should match snapshot 1`] = `
<div>
  <div
    className="navBar"
    data-testid="navbar"
  >
    <div
      className="leftNav"
    >
      <img
        alt="Logo"
        className="leftImage"
        onClick={[Function]}
        src="https://i.imgur.com/AHFcYv1.png"
      />
    </div>
    <div
      className="spacer"
    />
    <div
      className="middleNeighbor"
    >
      <img
        alt="Neighbor"
        className="siteFontLogo"
        src="https://i.imgur.com/pD5v9ZC.png"
      />
    </div>
    <div
      className="spacer"
    />
    <div
      className="rightNav"
    >
      <img
        alt="Team"
        className="teamImage"
        onClick={[Function]}
        src="https://i.imgur.com/FFrUPpK.png"
      />
      |
       
      <img
        alt="About Neighbors"
        className="aboutNeighborImage"
        onClick={[Function]}
        src="https://i.imgur.com/su5pf5g.png"
      />
      |
       
      <img
        alt="User Information"
        className="userInfo"
        onClick={[Function]}
        src="https://i.imgur.com/VWTc4IU.png"
      />
    </div>
  </div>
  <div>
    <h2
      className="login"
    >
      LoginUniqueWord
    </h2>
  </div>
  <div
    className="loginContainer"
  >
    <div
      className="consMerchContainer"
    >
      <div
        className="userLogin"
      >
        <div
          className="userAlignment"
        >
          <strong>
            User
          </strong>
        </div>
        <br />
        <form
          action="http://localhost:3000/userLogin"
          className="userForm"
          method="post"
        >
          <div
            className="userLogin"
          >
            <label
              htmlFor="email"
            >
              Email: 
            </label>
            <input
              id="email"
              name="email"
              type="text"
            />
          </div>
          <div
            className="userPW"
          >
            <label
              htmlFor="password"
            >
              Password: 
            </label>
            <input
              id="password"
              name="password"
              type="password"
            />
          </div>
          <input
            id="formButton"
            name="userSubmit"
            type="submit"
            value="submit"
          />
        </form>
      </div>
      <div
        className="merchantLogin"
      >
        <div
          className="merchantAlignment"
        >
          <strong>
            Merchant
          </strong>
        </div>
        <br />
        <form
          action="http://localhost:3000/merchantLogin"
          className="merchantForm"
          method="post"
        >
          <div
            className="merchantUsername"
          >
            <label
              htmlFor="memail"
            >
              Email: 
            </label>
            <input
              id="memail"
              name="memail"
              type="text"
            />
          </div>
          <div
            className="merchantPassword"
          >
            <label
              htmlFor="mpassword"
            >
              Password: 
            </label>
            <input
              id="mpassword"
              name="mpassword"
              type="password"
            />
          </div>
          <input
            id="merchantFormButton"
            type="submit"
            value="submit"
          />
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`NavBar component matches the snapshot 1`] = `
<div
  className="navBar"
  data-testid="navbar"
>
  <div
    className="leftNav"
  >
    <img
      alt="Logo"
      className="leftImage"
      onClick={[Function]}
      src="https://i.imgur.com/AHFcYv1.png"
    />
  </div>
  <div
    className="spacer"
  />
  <div
    className="middleNeighbor"
  >
    <img
      alt="Neighbor"
      className="siteFontLogo"
      src="https://i.imgur.com/pD5v9ZC.png"
    />
  </div>
  <div
    className="spacer"
  />
  <div
    className="rightNav"
  >
    <img
      alt="Team"
      className="teamImage"
      onClick={[Function]}
      src="https://i.imgur.com/FFrUPpK.png"
    />
    |
     
    <img
      alt="About Neighbors"
      className="aboutNeighborImage"
      onClick={[Function]}
      src="https://i.imgur.com/su5pf5g.png"
    />
    |
     
    <img
      alt="User Information"
      className="userInfo"
      onClick={[Function]}
      src="https://i.imgur.com/VWTc4IU.png"
    />
  </div>
</div>
`;

exports[`Team component matches the snapshot 1`] = `
<div
  className="teamPage"
  data-testid="team"
>
  <div>
    <div
      className="navBar"
      data-testid="navbar"
    >
      <div
        className="leftNav"
      >
        <img
          alt="Logo"
          className="leftImage"
          onClick={[Function]}
          src="https://i.imgur.com/AHFcYv1.png"
        />
      </div>
      <div
        className="spacer"
      />
      <div
        className="middleNeighbor"
      >
        <img
          alt="Neighbor"
          className="siteFontLogo"
          src="https://i.imgur.com/pD5v9ZC.png"
        />
      </div>
      <div
        className="spacer"
      />
      <div
        className="rightNav"
      >
        <img
          alt="Team"
          className="teamImage"
          onClick={[Function]}
          src="https://i.imgur.com/FFrUPpK.png"
        />
        |
         
        <img
          alt="About Neighbors"
          className="aboutNeighborImage"
          onClick={[Function]}
          src="https://i.imgur.com/su5pf5g.png"
        />
        |
         
        <img
          alt="User Information"
          className="userInfo"
          onClick={[Function]}
          src="https://i.imgur.com/VWTc4IU.png"
        />
      </div>
    </div>
  </div>
  <div
    className="allTeamPhotos"
  >
    <div
      className="zach"
    >
      <img
        className="teamPhoto"
        src="https://i.imgur.com/A0Q4L5l.png"
      />
      <div
        className="teamText"
      >
        <p>
          Zach Hall 
          <br />
          <br />
          Network Ninja
        </p>
        <a
          href="https://github.com/zachh85"
        >
          Github
        </a>
      </div>
    </div>
    <div
      className="chase"
    >
      <img
        className="teamPhoto"
        src="https://i.imgur.com/kXsSP3w.png"
      />
      <div
        className="teamText"
      >
        <p>
          Chase Benjamin 
          <br />
          <br />
          Digital Data Detective
        </p>
        <a
          href="https://github.com/chasebenj"
        >
          Github
        </a>
      </div>
    </div>
    <div
      className="danny"
    >
      <img
        className="teamPhoto"
        src="https://i.imgur.com/5JzRNX2.png"
      />
      <div
        className="teamText"
      >
        <p>
          Danny Zheng 
          <br />
          <br />
          Automation Wizard
        </p>
        <a
          href="https://github.com/dannyzheng159"
        >
          Github
        </a>
      </div>
    </div>
    <div
      className="sakura"
    >
      <img
        className="teamPhoto"
        src="https://i.imgur.com/DHJlkOx.png"
      />
      <div
        className="teamText"
      >
        <p>
          Sakura Akiyama-Bowden 
          <br />
          <br />
          Server Wrangler
        </p>
        <a
          href="https://github.com/sakurakiyama"
        >
          Github
        </a>
      </div>
    </div>
  </div>
</div>
`;
